sdk/textanalytics/azure-ai-textanalytics/swagger/authoring/README.md

```diff
@@ -0,0 +1,274 @@
+output-folder: ../../azure/ai/textanalytics/authoring
+title: TextAuthoringClient
+```
+
+## Customizations
+
+Customizations that should eventually be added to central autorest configuration.
+
+### General customizations
+
+```yaml
+directive:
+# Support automatically generating code for key credentials.
+- from: swagger-document
+  where: $.securityDefinitions
+  transform: |
+    $["AzureKey"] = $["apim_key"];
+    delete $["apim_key"];
+
+- from: swagger-document
+  where: $.security
+  transform: |
+    $ = [
+        {
+          "AzureKey": []
+        }
+    ];
+
+
+# Fix Endpoint parameter description and format.
+- from: swagger-document
+  where: $.parameters.Endpoint
+  transform: |
+    $["description"] = "Supported Cognitive Services endpoint (e.g., https://<resource-name>.cognitiveservices.azure.com).";
+    $["format"] = "url";
+```
+
+### Python customizations
+
+```yaml
+directive:
+# Always default to UnicodeCodePoint string indices.
+- from: swagger-document
+  where: $.definitions.StringIndexType
+  transform: |
+    $["description"] = "Specifies the method used to interpret string offsets. Set to \"UnicodeCodePoint\" for Python strings.";
+    $["x-ms-client-default"] = "UnicodeCodePoint";
+```
+
+### Authoring API Directives
+
+## Give LROs return types
+
+```yaml $(tag) == 'release_authoring_2022_05_01'
+directive:
+  - where-operation: TextAnalysisAuthoring_DeleteProject
+    transform: >
+      $["responses"]["200"] = {
+        "description": "dummy schema to get poller response when calling .result()",
+        "schema": {
+          "$ref": "#/definitions/TextAnalysisAuthoringProjectDeletionJobState"
+        }
+      };
+  - where-operation: TextAnalysisAuthoring_Export
+    transform: >
+      $["responses"]["200"] = {
+        "description": "dummy schema to get poller response when calling .result()",
+        "schema": {
+          "$ref": "#/definitions/TextAnalysisAuthoringExportProjectJobState"
+        }
+      };
+  - where-operation: TextAnalysisAuthoring_Import
+    transform: >
+      $["responses"]["200"] = {
+        "description": "dummy schema to get poller response when calling .result()",
+        "schema": {
+          "$ref": "#/definitions/TextAnalysisAuthoringImportProjectJobState"
+        }
+      };
+  - where-operation: TextAnalysisAuthoring_Train
+    transform: >
+      $["responses"]["200"] = {
+        "description": "dummy schema to get poller response when calling .result()",
+        "schema": {
+          "$ref": "#/definitions/TextAnalysisAuthoringTrainingJobState"
+        }
+      };
+  - where-operation: TextAnalysisAuthoring_DeployProject
+    transform: >
+      $["responses"]["200"] = {
+        "description": "dummy schema to get poller response when calling .result()",
+        "schema": {
+          "$ref": "#/definitions/TextAnalysisAuthoringDeploymentJobState"
```

Comment:
IIRC DeployProject is a PUT operation so the client library will follow the resourceLocation header to do a final GET on the original operation URL. In that case, this will actually return a `TextAnalysisAuthoringProjectDeployment`. Can you double-check on this behavior? Thanks!

https://github.com/Azure/azure-rest-api-specs/blob/main/specification/cognitiveservices/data-plane/Language/stable/2022-05-01/analyzetext-authoring.json#L2276